hwangsae/recorder.c

```diff
@@ -209,23 +212,30 @@ hwangsae_recorder_start_recording (HwangsaeRecorder * self, const gchar * uri)
   g_autoptr (GstBus) bus = NULL;
   g_autoptr (GstElement) element = NULL;
   g_autoptr (GstPad) parse_src = NULL;
+  g_autofree gchar *recording_edge_dir = NULL;
   g_autofree gchar *recording_file = NULL;
   g_autofree gchar *pipeline_str = NULL;
   const gchar *mux_name;
   g_autoptr (GError) error = NULL;
 
   g_return_val_if_fail (!priv->pipeline, ENOENT);
 
-  g_mkdir_with_parents (priv->recording_dir, 0750);
+  priv->edge_id = g_strdup (edge_id);
+
+  recording_edge_dir =
+      g_build_filename (priv->recording_dir, priv->edge_id, NULL);
+
+  g_mkdir_with_parents (recording_edge_dir, 0750);
```

Comment:
Thanks for your suggestion, I agree with you it is a better approach.

Reply:
I'm thinking that it would be useful to leave `edge_id` as parameter, to be able to search for a specific active `recording`  by edge_id. I will be probably useful in a near future. Do you agree?

Reply:
I think all the recorder should care about is the URI. Edge ID is an unnecessary detail, plus the recorder can save any compatible SRT stream, not only from Hwangsaeul edge.

I agree a mapping between edge_id and an active recording might be useful, but IMO this information should be kept at the level of HwansaeRecorderAgent (as some hash map or whatnot).

Reply:
Great. Thanks for sharing your thoughts.